packages/nimble-components/src/utilities/tests/storybook.ts

```diff
@@ -90,11 +90,8 @@ export const createFixedThemeStory = <TSource>(
                 <div
                     style="
                         background-color: ${backgroundState.value};
-                        position: absolute;
-                        width: 100%;
-                        min-height: 100%;
-                        left: 0px;
-                        top: 0px;
+                        margin: -1em;
```

Comment:
So my guess is this is trying to fight the `1rem` padding that storybook applies to the body?
![image](https://user-images.githubusercontent.com/1588923/220396583-1cbc5c33-123a-4b24-b37b-0518cf92c856.png)

If that is the goal maybe we should make that more explicit:
```ts
<style>
body {
    margin: 0px;
}
</style>
<nimble-theme-provider>
<div></div>
</nimble-theme-provider>
```

Guess I'm curious as to why though. Is there some requirement about the parent container that's needed in order to use the table (hopefully not)?

If it is to avoid a one-time chromatic diff for the fixedThemeStories maybe we can just do the diff?

A comment would be helpful describing it's pupose

Reply:
The requirement of the anchored region (and therefore the table) is that `document.documentElement` should appropriately reflect the size of the page. Otherwise, the anchored region can't correctly determine its placement (above/below, right/left) relative to its anchor element. To me, this seems like a reasonable requirement, but it isn't met within the current fixed theme stories because the only element in the document is absolutely positioned, which means the height of `document.documentElement` isn't an accurate representation of the page's size.

Regarding the actual approach I took to style the story, I went through a few different iterations of styling that all created issues with existing fixed theme stories. This seemed like the most reasonable approach, since it fixed the problems on all the stories and had the added benefit of not causing a huge diff. These are some of the approaches I tried:
1. Leave the `div` adding the background as absolutely positioned and make `${viewTemplate}` its sibling.
    a. **Problem:** This didn't allow the background to grow if the story was tall and had a scrollbar. The background was only behind the part of the story that was originally visible within the scrollable container.
2. Make the `div` relatively positioned with `${viewTemplate}` as its child. Have the height of the `div` be based on the height of the story.
    a. **Problem:** This caused problem for the dialog and drawer stories because the story itself had no height. Therefore, the background didn't fill the entire height of the page, which caused the dialog/drawer to not be shown on the correct background color.
3. Make the `div` relatively positioned with `${viewTemplate}` as its child. Have the min-height of the `div` be `100vh`.
   a. **Problem:** Every story had unnecessary scrollbars because `100vh` made the `div` too tall with the padding of the body.
4. Similar to (3) above, but set the min-height of the `div` be `calc(100vh - 2em)` to avoid the scrollbars. At which point, if the styling is making assumptions about the body styling, it might as well make the assumption in a way that avoids the large diff (i.e. set a negative margin rather than a height calculation).

As part of my investigation yesterday before I posted a PR, I briefly tried the approach you mentioned of adding a style for the `body`, but it didn't work. I didn't dig into it then, but looking at it again now, I see that it's because the story only returns the first child element, so I could write it like this:
```html
<nimble-theme-provider>
    <style>
        body {
            padding: 0px !important;
        }
    </style>
    <div></div>
</nimble-theme-provider>
```

Reply:
I can look for a place to document this -- perhaps for the table & for the menu-button itself.

I don't see anything that breaks with the style you suggested. It just means that all our fixed theme stories have a scrollbar.

Reply:
Thanks for the explanations. With that I don't have a strong preference. I personally find negative margins to be weird so I think I'd lean to the body padding: 0px example you posted above with a comment describing that it is overriding storybook's padding style.

I see the comments added to the table, menu-button, and action menu stories but I feel like they are too technical / nuanced for users to pick-up easily.

i.e.:
```
If the placement of the menu relative to the button is not configured using the \`position\` attribute, the menu's placement will
automatically be determined based on the size of \`document.documentElement\` and the menu button's location within it.
```

Does not convey to me needing to create CSS for the `<html>` element (I also expect most folks do not know that `document.documentElement` corresponds to the `<html>` tag), what a recommended configuration of that is, or what they need to look out for in the page if things are not positioned correctly. I think it's a pretty nuanced discussion.

This seems like a fundamental set-up / integration behavior that needs to be clearly defined per framework in their READMEs.

I don't think it should be addressed in this PR, I think it'll take a couple of rounds of discussion. But can you create a tech debt issue so we can capture it?

Reply:
Created #1084